pkg/p2p/libp2p/internal/handshake/handshake.go

```diff
@@ -144,7 +144,7 @@ func (s *Service) Handshake(ctx context.Context, stream p2p.Stream, peerMultiadd
 		return nil, fmt.Errorf("read synack message: %w", err)
 	}
 
-	observedUnderlay, err := ma.NewMultiaddrBytes(resp.Syn.ObservedUnderlay)
+	observedUnderlay, err := ma.NewMultiaddrBytes(fullRemoteMABytes)
```

Comment:
ObservedUnderlay exists to communicate the real underlay that a peer is connected to another peer. This may be different than the fullRemoteMA if the peer is under NAT. Hence the name observable. I do not think that this change is correct as it makes the ObservedUnderlay irrelevant and could be removed completely. With this change observedUnderlay and fullRemoteMA are always the same and can be even more simplified, but with losing the information about the observed underlay by the peer.

If there is investigation that observable underlay is not needed here or in the Handle method, it would be good to add to the PR description.

Reply:
Thanks, I will add the relevant info to the description. The problem we are trying to tackle is that `Syn.ObservableUnderlay` value belongs to the unsigned part of the handshake payload and can be manipulated. With this PR we are trying to mitigate this risk.

Reply:
I am suggesting to sign observable underly instead to just ignore it. Or, at least, to analyze the impact of removing this information.

Reply:
After the discussion we had we decided to put a warning in place.